### ISSUE TITLE
Is this app going to be dead?

### ISSUE THREAD

**Issue by echoaldemo** [Aldrin-John-Olaer-Manalansan/DOTA-2-MOD-Master#37]
Dota 2 now recognizes any changes made to gameinfo.gi as cheating.\

**Comment by LucazsD** [comment]
We probably have to wait for the author to bypass VAC, gonna take a long time, or perhaps never. Anyway, Are there any mod programs that don't modify the game directory ???

**Comment by LucazsD** [comment]
Plus, devs seem to change a lot of items' names and places as well. A lot of not-found errors pop-ups while injecting.

**Comment by ducbh62git** [comment]
I've met this problem today. Now we got a "Valve anti cheat" pop-up when clicking play button

**Comment by smackdownn** [comment]
is it possible to modify the game files directly? In the reddit post it is mentioned that modifying gameinfo.gi brings up the vac error. What if the original vpk itself is modified instead of using a mods folder, will it work?

**Comment by Aldrin-John-Olaer-Manalansan** [comment]
> is it possible to modify the game files directly? In the reddit post it is mentioned that modifying gameinfo.gi brings up the vac error. What if the original vpk itself is modified instead of using a mods folder, will it work?


There is **no** application/method out there that allows users to **directly modify the original pak01_dir.vpk**. The idea was actually great and if it's possible then that will be revolutionary throughout the modding community.






> Dota 2 now recognizes any changes made to gameinfo.gi as cheating.\

Sadly without **gameinfo.gi** DOTA2 Modding will not be possible anymore. For the mean time all users should stop MODDING your DOTA2 for your own account's Safety.

**it's sad to say this but DOTA2 MOD Master will momentarily shutdown** until the community found a new way of Modding DOTA2.

**Comment by tyzmodo** [comment]
There are currently two ways to mod dota after that vac warning,one of them is suitable to most players,let me explain it below:
1.create a folder under your game directory named dota_xxx(replace 'xxx' with your custom mod name)
2.create a simpler version of gameinfo.gi as this:
`"GameInfo"
{
	LayeredOnMod	dota

	FileSystem
	{
		SearchPaths
		{
			Game				dota_xxx(replace 'xxx' with that folder name above)
			Game				dota
			Game				core

			Mod					dota_xxx(replace 'xxx' with that folder name above)
			Mod					dota

			AddonRoot			dota_addons

			// Note: addon content is included in publiccontent by default.
			PublicContent		core
		}
	}
}
`
3.put your mod pak01_dir.vpk into the folder xxx (replace 'xxx' with the folder name you provided )
4. now add -language xxx (replace xxx with folder name above) launch option to dota and start dota then you should see the mod effects and no more vac warning.

**Comment by ducbh62git** [comment]
I've implemented your solution. It worked but some skins work wrong and all skins have no effect at all. My setup as below:

1. create  `dota_Aldrin_Mods` folder in game directory
2. Add `gameinfo.gi` as below:

```
"GameInfo"
{
	LayeredOnMod	dota

	FileSystem
	{
		SearchPaths
		{
			Game				dota_Aldrin_Mods
			Game				dota
			Game				core

			Mod					dota_Aldrin_Mods
			Mod					dota

			AddonRoot			dota_addons

			// Note: addon content is included in publiccontent by default.
			PublicContent		core
		}
	}
}
```
3. Copy all files and folders from previous mod folder to `dota_Aldrin_Mods`
4. add `-language Aldrin_Mods` to lauch option

Here is the error example on Phantom Assassin:
<img width="1280" alt="error" src="https://user-images.githubusercontent.com/69949959/149657910-7efcc970-3c86-41e1-afb6-1b8ef5d45413.png">

**Comment by tyzmodo** [comment]
thanks for pointing out @ducbh62git

**Comment by ducbh62git** [comment]
So I think before fixing the solution1 error. Can you please provide the second solution ? @tyzmodo

**Comment by tyzmodo** [comment]
the second one is not suitable for all players,but I'll provide it anyway :

**Comment by ducbh62git** [comment]
Do you use Aldrin_Mods ?

**Comment by tyzmodo** [comment]
I used it before but not recently,now i keep an eye on this tool

**Comment by ducbh62git** [comment]
Yah, my `pak01_dir.vpk` is now 137 MB estimately. So I don't think it's the problem. 
Please share the way to implement the second solution.

**Comment by tyzmodo** [comment]
ok sorry i thought i said it clearly,let me explain it in detail:
1.You need to be a perfectworld server player in lowviolence mode (meaning there's a dota_lv folder in your game directory)
2.then there are two ways of modding you can choose (1 delete the vpks in dota_lv and put your pak01_dir.vpk into dota_lv. (2 unpack pak01_dir.vpk in dota_lv and mix the contents of your mod files with it then regenerate a pak01_dir.vpk
That's all

**Comment by tyzmodo** [comment]
but i have to point out that sound mod doesn't work well using the second method

**Comment by chall2** [comment]
Guys there is an easier way. You can simply adjust the size of the CRC to suit your needs. The number of characters in the file is also checked; it must be the same size as the original file .gi 

I have already implemented this in my changer https://vk.com/dota2reload

Use this tool: https://github.com/rr-/CRC-manipulator

**Comment by azevedus** [comment]
@chall2chall4 method works like a charm!

Just had to delete a few characters of the commentary lines on the gameinfo.fi to make sure the final size after CRC patching matched the original gameinfo.fi.

**Comment by smackdownn** [comment]
the above solution works but now aldrin mod master is throwing a lot of "cannot find dota_npc_hero_xxxx" errors. I got like 15 such errors which included phantom assassin, earthshaker, venomancer, undying etc. This didnt happen before.

**Comment by faheemiq** [comment]
Does this solution still work? I have just tried it and it gives VAC game files modified error.

**Comment by Aldrin-John-Olaer-Manalansan** [comment]
VAC notification has now been fixed, update DOTA2 MOD Master to v2.10.0 .
